launchflow/gcp/firebase.py

```diff
@@ -0,0 +1,73 @@
+import dataclasses
+from typing import List, Optional
+
+from launchflow.gcp.resource import GCPResource
+from launchflow.models.enums import ResourceProduct
+from launchflow.models.flow_state import EnvironmentState
+from launchflow.node import Depends, Outputs
+from launchflow.resource import ResourceInputs
+
+
+@dataclasses.dataclass
+class FirebaseProjectOutputs(Outputs):
+    pass
+
+
+@dataclasses.dataclass
+class FirebaseProjectInputs(ResourceInputs):
+    pass
+
+
+class FirebaseProject(GCPResource[FirebaseProjectOutputs]):
+    product = ResourceProduct.GCP_FIREBASE_PROJECT.value
+
+    def __init__(self, name: str) -> None:
+        """Create a new Firebase Project resource.
+        **Args:**
+        - `name (str)`: The name of the Firebase project.
+        """
+        super().__init__(name=name)
+        # public metadata
+
+    def inputs(self, environment_state: EnvironmentState) -> FirebaseProjectInputs:
+        return FirebaseProjectInputs(resource_id=self.resource_id)
+
+
+@dataclasses.dataclass
+class FirebaseHostingSiteOutputs(Outputs):
+    default_url: str
+    desired_dns_records: List[str]
+
+
+@dataclasses.dataclass
+class FirebaseHostingSiteInputs(ResourceInputs):
+    firebase_project_id: str
+    custom_domain: Optional[str]
```

Comment:
nit: maybe just `domain` to match the other service

Reply:
Left a TODO, Im gonna do a general sweep through these resources again with the s3 static site change tomorrow, will address this there.